src/types.ts

```diff
@@ -168,3 +169,11 @@ export type ProtocolStream = NoiseStream & { userData: Protomux }
 export type Entries<T> = {
   [K in keyof T]: [K, T[K]]
 }[keyof T][]
+
+export type ProjectKeys = {
+  projectKey: Buffer
+  projectSecretKey?: Buffer
+  authEncryptionKey: Buffer
+} & {
+  [n in `${Exclude<(typeof NAMESPACES)[number], 'auth'>}EncryptionKey`]?: Buffer
+}
```

Comment:
See renaming comment above - you can reduce a lot of typing with:

```suggestion
import { SetRequired } from 'type-fest'

export type ProjectKeys = {
  projectKey: Buffer
  projectSecretKey?: Buffer
  encryptionKeys: SetRequired<{
    [typeof NAMESPACES)[number]]?: Buffer
  }, 'auth'>
}
```

Reply:
addressed via 6d14bacb6f178af5102c8d855efffbb0d3b4c085